core/src/main/java/com/google/cloud/sql/core/DefaultConnectionInfoCache.java

```diff
@@ -35,7 +35,7 @@
 class DefaultConnectionInfoCache {
   private final AccessTokenSupplier accessTokenSupplier;
   private final CloudSqlInstanceName instanceName;
-  private final RefreshAheadStrategy refreshStrategy;
+  private final RefreshStrategy refreshStrategy;
```

Comment:
should the default not still be `RefreshAheadStrategy`?

Reply:
Technically yes, it could because there is exactly one implementation. But because this PR is about introducing the RefreshStrategy interface class, I think it makes sense update the code where the interface gets used together.